core/src/main/java/com/datastax/oss/driver/internal/core/resolver/mockResolver/MockResolver.java

```diff
@@ -0,0 +1,83 @@
+package com.datastax.oss.driver.internal.core.resolver.mockResolver;
+
+import com.datastax.oss.driver.internal.core.resolver.Resolver;
+import com.datastax.oss.driver.internal.core.resolver.Response;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.UnknownHostException;
+
+/** Resolver implementation that allows returning custom results. */
+public class MockResolver implements Resolver {
+  private final MockResolverResultSource resultSource;
+
+  MockResolver(MockResolverResultSource resultSource) {
+    this.resultSource = resultSource;
+  }
+
+  /**
+   * Returns first {@link InetAddress} from predefined {@link Response} or forwards the call to
+   * {@link InetAddress#getByName(String)} if undefined.
```

Comment:
```suggestion
   * Returns first {@link InetAddress} from overrided results {@link Response} if one is present.
   * Otherwise forwards the call to {@link InetAddress.getByName}
```